tests/zoltan/zoltan_02.cc

```diff
@@ -0,0 +1,88 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2004 - 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// check GridTools::color_sparsity_pattern
+
+
+#include "../tests.h"
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_iterator.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/lac/sparsity_tools.h>
+#include <deal.II/numerics/data_out.h>
+
+#include <fstream>
+#include <vector>
+
+
+void fill_graph (DynamicSparsityPattern &graph)
+{
+  //Edges in only one direction
+  graph.add(0,1);
+  graph.add(0,2);
+  graph.add(1,2);
+  graph.add(1,4);
+  graph.add(4,3);
+  graph.add(2,4);
+  graph.add(3,2);
+
+  //Edges in opposite direction
+  graph.add(1,0);
+  graph.add(2,0);
+  graph.add(2,1);
+  graph.add(4,1);
+  graph.add(3,4);
+  graph.add(4,2);
+  graph.add(2,3);
+}
+
+
+int main (int argc, char **argv)
+{
+  initlog();
```

Comment:
look at other tests with `mpi` and please extend this one to run with more than one MPI core. That's done by adding another output file `zoltan_02.mpirun=3.output` and adding a few things here. Look at `tests/lac/linear_operator_14.cc`, for example.